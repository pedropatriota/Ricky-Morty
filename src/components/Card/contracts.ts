export interface ICardProps {
	name: string;
	species: string;
	status: string;
	gender: string;
	image: string;
}
