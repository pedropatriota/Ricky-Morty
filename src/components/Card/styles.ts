import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
