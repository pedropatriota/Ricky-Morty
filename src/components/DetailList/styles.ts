import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
