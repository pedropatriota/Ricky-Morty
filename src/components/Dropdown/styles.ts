import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
