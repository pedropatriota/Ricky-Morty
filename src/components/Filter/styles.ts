import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-end;
	justify-content: space-between;
	flex-wrap: wrap;
	gap: 0.8rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
