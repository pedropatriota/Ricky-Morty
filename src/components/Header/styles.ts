import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
