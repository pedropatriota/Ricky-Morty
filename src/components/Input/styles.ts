import styled from 'styled-components';

export const Input = styled.input`
  width: min(90vw, 30rem);
  border: ${(props) => props.theme.colors.border};
  color: ${(props) => props.theme.colors.text};
  height: 3.375rem;
  border-radius: 3px;
  padding: 0 2rem;
  font-size:0.94rem;
  outline: none;
  background: ${(props) => props.theme.colors.el}
`;
