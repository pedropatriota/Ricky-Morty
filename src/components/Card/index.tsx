import { renderData } from '../../utils';
import type { ICardProps } from './contracts';
import * as Styled from './styles';

const Card = ({ name, image, species, gender, status }: ICardProps) => {
	return (
		<Styled.Container data-testid="card">
			<Styled.ImagContainer image={image} data-testid="card-image" />
			<Styled.InfoContainer>
				<h2>{name}</h2>
				{renderData('Status', status)}
				{renderData('Species', species)}
				{renderData('Gender', gender)}
			</Styled.InfoContainer>
		</Styled.Container>
	);
};

export default Card;
