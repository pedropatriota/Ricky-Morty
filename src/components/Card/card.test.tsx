import { screen, render, within } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { light } from '../../styles/themes';
import Card from '.';

describe('Card', () => {
	const props = {
		gender: 'Male',
		image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
		name: 'Rick Sanchez',
		species: 'Human',
		status: 'Alive'
	};
	it('Should render each Card correctly', async () => {
		render(
			<ThemeProvider theme={light}>
				<Card {...props} />
			</ThemeProvider>
		);

		const card = screen.getByTestId('card');
		const image = screen.getByTestId('card-image');
		const name = screen.getByRole('heading', {
			name: /rick sanchez/i
		});

		const status = within(card).getByText(/alive/i);
		const gender = within(card).getByText(/male/i);
		const specie = within(card).getByText(/human/i);

		expect(card).toBeInTheDocument();
		expect(image).toBeInTheDocument();
		expect(name).toBeInTheDocument();
		expect(status).toBeInTheDocument();
		expect(gender).toBeInTheDocument();
		expect(specie).toBeInTheDocument();
	});

	it('should match the snapshot', () => {
		const { container } = render(
			<ThemeProvider theme={light}>
				<Card {...props} />
			</ThemeProvider>
		);
		expect(container).toMatchSnapshot();
	});
});
