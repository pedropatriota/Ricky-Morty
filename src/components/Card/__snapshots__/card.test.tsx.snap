// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
<div>
  <div
    class="sc-aXZVg bNflSf"
    data-testid="card"
  >
    <div
      class="sc-gEvEer cvOKXx"
      data-testid="card-image"
    />
    <div
      class="sc-eqUAAy jAJozm"
    >
      <h2>
        Rick Sanchez
      </h2>
      <p>
        <strong>
          Status
          : 
        </strong>
         
        Alive
      </p>
      <p>
        <strong>
          Species
          : 
        </strong>
         
        Human
      </p>
      <p>
        <strong>
          Gender
          : 
        </strong>
         
        Male
      </p>
    </div>
  </div>
</div>
`;
